specification/cognitiveservices/data-plane/TextAnalytics/preview/v3.1-preview.1/TextAnalytics.json

```diff
@@ -0,0 +1,1346 @@
+        },
+        "confidenceScores": {
+          "description": "Aspect level sentiment confidence scores for the aspect in the sentence.",
+          "$ref": "#/definitions/AspectConfidenceScoreLabel"
+        },
+        "offset": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The aspect offset from the start of the sentence."
+        },
+        "length": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The length of the aspect by Unicode standard."
+        },
+        "text": {
+          "type": "string",
+          "description": "The aspect text detected."
+        },
+        "relations": {
+          "type": "array",
+          "description": "The array of either opinion or aspect object which is related to the aspect.",
+          "items": {
+            "$ref": "#/definitions/AspectRelation"
+          }
+        },
+        "warnings": {
+          "type": "array",
+          "description": "The warnings generated for the sentence.",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "SentenceOpinion": {
+      "type": "object",
+      "required": [ "confidenceScores", "id", "isNegated", "length", "offset", "sentiment", "text" ],
+      "properties": {
+        "id": {
+          "type": "string",
+          "description": "Unique, non-empty document identifier."
+        },
+        "sentiment": {
+          "type": "string",
+          "enum": [ "positive", "mixed", "negative" ],
+          "description": "Opinion level sentiment for the aspect in the sentence."
+        },
+        "confidenceScores": {
+          "description": "Opinion level sentiment confidence scores for the aspect in the sentence.",
+          "$ref": "#/definitions/AspectConfidenceScoreLabel"
+        },
+        "offset": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The opinion offset from the start of the sentence."
+        },
+        "length": {
+          "type": "integer",
+          "format": "int32",
+          "description": "The length of the opinion by Unicode standard."
+        },
+        "text": {
+          "type": "string",
+          "description": "The aspect text detected."
+        },
+        "isNegated": {
+          "type": "boolean",
+          "description": "The indicator representing if the opinion is negated."
+        },
+        "warnings": {
+          "type": "array",
+          "description": "The warnings generated for the sentence.",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "AspectRelation": {
+      "type": "object",
+      "required": [ "links", "relationType" ],
+      "properties": {
+        "relationType": {
+          "type": "string",
+          "enum": [ "opinion", "aspect" ],
+          "description": "The type related to the aspect."
+        },
+        "links": {
```

Comment:
`ref` or `$ref`

Reply:
This one... still I dont know. please refer to my other reply. I think we need discussion on this.

Reply:
I changed to "ref" :)